packages/terra-application-layout/examples/About.site-page.jsx

```diff
@@ -0,0 +1,22 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import React from 'react';
+import PropsTable from 'terra-props-table';
+import Markdown from 'terra-markdown';
+import AboutDoc from '../docs/about.md';
+import { version } from '../package.json';
+
+// Component Source
+/* eslint-disable import/no-webpack-loader-syntax, import/first, import/no-unresolved, import/extensions */
+import ApplicationLayoutSrc from '!raw-loader!../src/ApplicationLayout.jsx';
+/* eslint-enable import/no-webpack-loader-syntax, import/first, import/no-unresolved, import/extensions */
+
+const About = () => (
+  <div>
+    <div id="version">Version: {version}</div>
+    <Markdown id="about" src={AboutDoc} />
+    <br />
+    <PropsTable id="props-header" src={ApplicationLayoutSrc} componentName="ApplicationLayout" />
```

Comment:
I tried to use the application-layout component from just reading the docs similar to how our users will go about using this component. Looking at the prop table, all of the props are listed as optional except for `indexPath`, but I'm finding out that all props are actually required. If I remove any of the props in `ExampleApplication.jsx`, I get a console error around how some function is trying to use data from one of the removed props. I also noticed that most of the prop type values are listed as custom, and the actual shape of them is not clear in this prop table.

I'd recommend removing the `<PropsTable />` component altogether for this component as it doesn't really provide much helpful information other than what props there are and their description. I'd replace the props table here with rendering the `usage.md` doc and drop the `packages/terra-application-layout/examples/Usage.site-page.jsx` page. And then in the `usage.md` file where the props are more clearly documented, I'd add a note to each prop about it being required.

Reply:
I think that makes sense. Just to clarify, the only truly required props are the routingConfig and the indexPath. The others may make sense to mix and max in some circumstances. I'll update the proptypes to specify those as required.

There were a couple bugs introduced in some refactoring that caused a missing nameConfig to throw exceptions, but those have been fixed.

I'll get the usage included in About page.